video/out/w32_common.c

```diff
@@ -1315,6 +1315,29 @@ static void update_cursor_passthrough(const struct vo_w32_state *w32)
     }
 }
 
+static void update_native_touch(const struct vo_w32_state *w32)
+{
+    if (w32->parent)
+        return;
+
+    // Block unwanted gestures (press and hold etc)
+    GESTURECONFIG gc = {0};
+    if (w32->opts->native_touch) {
+        RegisterTouchWindow(w32->window, 0);
+        gc.dwBlock = GC_ALLGESTURES;
+    } else {
+        UnregisterTouchWindow(w32->window);
+        gc.dwWant = GC_ALLGESTURES;
+    }
+    SetGestureConfig(w32->window, 0, 1, &gc, sizeof(gc));
+}
+
+// If native touch is enabled and the left mouse button event is emulated, ignore it.
+static bool should_ignore_mouse_event(const struct vo_w32_state *w32)
+{
+    return w32->opts->native_touch && ((GetMessageExtraInfo() & 0xFFFFFF00) == 0xFF515700);
```

Comment:
consider copying the definitions from [here](https://learn.microsoft.com/en-us/windows/win32/tablet/system-events-and-mouse-messages#distinguishing-pen-input-from-mouse-and-touch) to make it self-explanatory *or* put a link